packages/strapi/lib/middlewares/router/index.js

```diff
@@ -52,27 +50,18 @@ module.exports = strapi => {
           });
 
           // Exclude routes with prefix.
-          const excludedRoutes = _.omitBy(
-            plugin.config.routes,
-            o => !_.has(o.config, 'prefix')
-          );
-
-          _.forEach(
-            _.omit(plugin.config.routes, _.keys(excludedRoutes)),
-            value => {
+          (plugin.config.routes || [])
+            .filter(route => !_.has(route.config, 'prefix'))
+            .forEach(value => {
               composeEndpoint(value, { plugin: pluginName, router });
-            }
-          );
+            });
 
-          // /!\ Could override main router's routes.
-          if (!_.isEmpty(excludedRoutes)) {
-            _.forEach(excludedRoutes, value => {
-              composeEndpoint(value, {
-                plugin: pluginName,
-                router: strapi.router,
-              });
+          // if you set a prefix key in the route it will ignore the plugin prefix
+          (plugin.config.routes || [])
+            .filter(route => _.has(route.config, 'prefix'))
+            .forEach(value => {
+              composeEndpoint(value, { plugin: pluginName, router: strapi.router });
```

Comment:
Why duplicate almost identical code instead of doing something like

``` javascript
(plugin.config.routes || [])
    .forEach(value => {
        const hasPrefix = _.has(route.config, 'prefix');
        composeEndpoint(value, {
            plugin: pluginName,
            router: hasPrefix ? strapi.router : router
        });
    });
```

Reply:
sure yes